Realm/RLMArrayLinkView.mm

```diff
@@ -194,12 +200,13 @@ - (void)replaceObjectAtIndex:(NSUInteger)index withObject:(RLMObject *)object {
 - (NSUInteger)indexOfObject:(RLMObject *)object {
     // check attached for table and object
     RLMLinkViewArrayValidateAttached(self);
+    RLMValidateObjectClass(object, self.objectClassName);
     if (object->_realm && !object->_row.is_attached()) {
         @throw RLMException(@"RLMObject is no longer valid");
     }
 
     // check that object types align
-    if (![_objectClassName isEqualToString:object.objectSchema.className]) {
+    if (![_objectClassName isEqualToString:object->_objectSchema.className]) {
```

Comment:
Although note that the error message in RLMValidateObjectClass is wrong for this case.